libyul/optimiser/Suite.cpp

```diff
@@ -172,11 +172,15 @@ void OptimiserSuite::run(
 	Rematerialiser::run(*_dialect, ast);
 	UnusedPruner::runUntilStabilised(*_dialect, ast, reservedIdentifiers);
 
-	(FunctionGrouper{})(ast);
+	FunctionGrouper{}(ast);
 	StackCompressor::run(_dialect, ast);
-	(BlockFlattener{})(ast);
+	BlockFlattener{}(ast);
 
+	FunctionHoister{}(ast);
+	BlockFlattener{}(ast);
 	VarNameCleaner{ast, *_dialect, reservedIdentifiers}(ast);
+	FunctionGrouper{}(ast);
+	StackCompressor::run(_dialect, ast);
```

Comment:
StackCompressor modifies the ast and I would not want to do that for non-disambiguated code. Better to just run the CompilabilityChecker.

Reply:
Running AsmAnalyzer now after VarNameCleaner